### PR TITLE
feat(docs): FRMW-160 - Migrate rxApp docs in README to ngdocs

### PR DESCRIPTION
**Pull request by bry4n**
JIRA: https://jira.rax.io/browse/FRMW-160

* [x] Dev LGTM
* [x] Dev LGTM

### CODE REVIEW

File: src/rxApp/rxApp.js [hunk 1]
```diff
@@ -4,7 +4,662 @@
...
+ * <rx-page title="'Example Page'">
+ *    Here is my content
+ * </rx-page>
+ * </pre>
+ *
+ * `rx-page` is used to create a common wrapper for specific page views. It
+ * automatically adds the breadcrumbs and page title/subtitle (if specified),
+ * along with the correct styling.
+ *
+ * ( Both the `title` and `subtitle` attributes accept an Angular expression,
```
Comment (100stacks) [inline src/rxApp/rxApp.js]: Looks like a extra **`(`**

File: src/rxApp/README.md [hunk 2]
```diff
@@ -1,464 +1,3 @@
 [![unstable](http://badges.github.io/stability-badges/dist/unstable.svg)](http://github.com/badges/stability-badges)
 
-# Description
-
 This component is responsible for creating the HTML necessary for a common Encore layout. It builds out the main navigation, plus breadcrumbs and page titles.
```
Comment (100stacks) [inline src/rxApp/README.md]: As a major component of the Encore Framework, should the description include something such as a `major component` tag/label?  Simply a suggestion, which may change after the IA rewrite.

File: src/rxApp/rxApp.js [hunk 3]
```diff
@@ -4,7 +4,662 @@
...
+ * separate resource. This source can be changed, and there are many options to
+ * control the navigation from an app level.
+ *
+ * ## Accessing route information
+ *
+ * Sometimes it's helpful to have the current route information available for
+ * menu items. For example, re-using the current params for path building.
+ *
+ * To help with this, $route is exposed on the scope of all menu items.
+ * [`$route` provides many details on the current view](http://devdocs.io/angular/ngroute.$route),
```
Comment (100stacks) [inline src/rxApp/rxApp.js]: Link has changed.  `http://devdocs.io/angular/ngroute/service/$route`

File: src/rxApp/rxApp.js [hunk 4]
```diff
@@ -4,7 +4,662 @@
...
+ * demo shows the use of HTML tags.
+ *
+ * In either case (`title` or `unsafe-html-title`), the document title
+ * (i.e. visible in the browser tab) will be set to your chosen title. If you use
+ * `unsafe-html-title`, all HTML tags will be stripped before setting the document
+ * title.
+ *
+ * ### Account Info below Breadcrumbs
+ *
+ * `rxPage` integrates the [rxAccountInfo](#/components/rxAccountInfo) component,
```
Comment (100stacks) [inline src/rxApp/rxApp.js]: These shortened links (e.g., `#/components/rxAccountInfo`) currently do not work from our ngdocs API documentation pages.  I had the same issue.  Can we add a new route to handle them?

File: src/rxApp/README.md (hunk 2 above)
Comment (CITguy) [inline src/rxApp/README.md]: I'm fine with this description for now. I agree that the IA updates may change this anyway.

**Comment by CITguy** [comment]
This might be one case where the README docs will need to be split amongst the `rxApp` overview page and each directive/service/etc. defined within `rxApp`. Please arrange documentation accordingly.

File: src/rxApp/rxApp.js (hunk 4 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: fixed

File: src/rxApp/rxApp.js (hunk 1 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: fixed

File: src/rxApp/rxApp.js (hunk 4 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: nevermind... it isn't fixed.

File: src/rxApp/rxApp.js [hunk 5]
```diff
@@ -4,7 +4,662 @@
...
+ * tags are drawn inside of a `<span>`, essentially as:
+ *
+ * <pre>
+ * <span class="status-tag {{ class }}">{{ text }}</span>
+ * </pre>
+ *
+ * To use this, do the following in your application's `.config()` method:
+ *
+ * <pre>
+ * rxStatusTagsProvider.addStatus({ key: 'gamma', class: 'alpha-status', text: 'Hello World!' });
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: This is challenging to skim when generated...

screen shot 2015-09-23 at 3 40 12 pm


Please place key/value pairs on separate lines.

```javascript
rxStatusTagsProvider.addStatus({
    key: 'gamma',
    class: 'alpha-status',
    text: 'Hello World!'
});
```

File: src/rxApp/rxApp.js [hunk 6]
```diff
@@ -4,7 +4,662 @@
...
+ * <pre>
+ * <rx-page title="'Some Title'" status="gamma">
+ * </pre>
+ *
+ * And the title will appear with a `Hello World!` tag beside it, styled the
+ * same way as our `"alpha"` status tag is styled. You can also define your own
+ * CSS style in your application and use those instead, passing it as the `class`
+ * value to `addStatus()`.
+ *
+ * All the tags are accessible inside of [rxBreadcrumbs](/#/components/rxBreadcrumbs)
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Modify link to point to `rxBreadcrumbs` API page (`{@link rxBreadcrumbs}`).

File: src/rxApp/rxApp.js [hunk 7]
```diff
@@ -4,7 +4,662 @@
...
+ * demo shows the use of HTML tags.
+ *
+ * In either case (`title` or `unsafe-html-title`), the document title
+ * (i.e. visible in the browser tab) will be set to your chosen title. If you use
+ * `unsafe-html-title`, all HTML tags will be stripped before setting the document
+ * title.
+ *
+ * ### Account Info below Breadcrumbs
+ *
+ * `rxPage` integrates the [rxAccountInfo](/#/components/rxAccountInfo) component,
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Grammar needs some correction, and link needs to point to API overview for `rxAccountInfo`.

```
`rxPage` integrates with the {@link rxAccountInfo} component,
```

File: src/rxApp/rxApp.js [hunk 8]
```diff
@@ -4,7 +4,662 @@
...
+ * to draw the Account Info box directly underneath the `rxBreadcrumbs`.
+ * This is opt-in. By default, it will not appear. To enable it, pass the
+ * `account-number="..."` attribute to `<rx-page>` in your template, i.e
+ *
+ * <pre>
+ * <rx-page account-number="{{ accountNumber }}">
+ * </pre>
+ *
+ * As noted on the [rxAccountInfo](/#/components/rxAccountInfo) demo page, this
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Link needs updated and wording no longer makes sense.

```
As noted in {@link rxAccountInfo}, this ...
```

File: src/rxApp/rxApp.js [hunk 9]
```diff
@@ -4,7 +4,662 @@
...
+ * `accountNumber` to correctly operate.
+ *
+ * ### `childHeader` (optional)
+ *
+ * The `childHeader` attribute is used to specify an HTML header to be placed
+ * above the `children` in an expanded area (and thus having a `childHeader`
+ * attribute requires having a `children` attribute).
+ *
+ * `childHeader` receives HTML content as a string, and uses
+ * [rxCompile](#/components/rxCompile) to compile and insert the content above
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Link needs updated to point to API docs.

```
{@link rxCompile} to compile ...
```

File: src/rxApp/rxApp.js [hunk 10]
```diff
@@ -4,7 +4,662 @@
...
+ *    </div>
+ *    <img src="http://cdn.memegenerator.net/instances/500x/48669250.jpg"
+ *         alt="Look at all these servers there are so many" />
+ * </rx-page>
+ * </pre>
+ *
+ * ## rx-app-nav and rx-app-nav-item
+ *
+ * These two directives are responsible for creating the menu in the left sidebar.
+ * They're not intended for use outside of the rx-app template code.
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Please remove this section. If the directives are not meant to be used outside of `rx-app` template code, there's no reason to bring them up in the overview.

File: src/rxApp/rxApp.js [hunk 11]
```diff
@@ -4,7 +4,662 @@
...
+ * the `rx-app` directive via the `menu` attribute. View the demo for an example
+ * of this.
+ *
+ * # Sub-directives
+ *
+ * ## rx-page
+ *
+ * You'll likely want to use `rx-page` inside your template view. For example,
+ * inside a 'myView.html' file:
+ *
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: The above paragraph should read:

```
You'll likely want to use the {@link rxApp.directive:rxPage rxPage} directive inside your template view.
```

Please move documentation (START to END) to `rxPage` directive documentation.

START: `rxPage`

File: src/rxApp/rxApp.js [hunk 12]
```diff
@@ -4,7 +4,662 @@
...
+ * <pre>
+ * <rx-page title="'Servers Overview'">
+ *    <div class="page-actions">
+ *        <a href="/create" class="link-action msg-action">Create New Server</a>
+ *    </div>
+ *    <img src="http://cdn.memegenerator.net/instances/500x/48669250.jpg"
+ *         alt="Look at all these servers there are so many" />
+ * </rx-page>
+ * </pre>
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: END: `rxPage` directive documentation

File: src/rxApp/rxApp.js [hunk 13]
```diff
@@ -4,7 +4,662 @@
...
+ *
+ * The EncoreUI default font is Roboto. This is used for all text on the page
+ * and is loaded via Google Fonts. Be sure your app includes the following line:
+ *
+ * <pre>
+ * <link href="https://fonts.googleapis.com/css?family=Roboto:400,100,100italic,300,300italic,400italic,700,700italic"
+ *       rel="stylesheet" type="text/css" />
+ * </pre>
+ *
+ * # Navigation Menu JSON Structure
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Documentation about navigation menu JSON structure should be placed in `rx-app-nav` directive documentation.

File: src/rxApp/rxApp.js (hunk 12 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: moved to rxPage directive

File: src/rxApp/rxApp.js (hunk 13 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: moved to rxAppNav directive

File: src/rxApp/rxApp.js (hunk 10 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: removed

File: src/rxApp/rxApp.js (hunk 9 above)
Comment (bry4n) [inline src/rxApp/rxApp.js]: updated

**Comment by 100stacks** [comment]
LGTM

**Comment by CITguy** [comment]
LGTM, though I'm not entirely happy with the location of the nav JSON documentation, but we can address that later.

File: src/rxApp/rxApp.js [hunk 14]
```diff
@@ -282,6 +470,469 @@ angular.module('encore.ui.rxApp', [
...
+ *                 "key": "overview",
+ *                 "linkText": "Overview"
+ *           }, {
+ *                 "href": "/about",
+ *                 "key": "about",
+ *                 "linkText": "About"
+ *           },
+ *       ]
+ * }
+ * <pre>
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: This should be closing `</pre>`

Comment (bry4n) [inline src/rxApp/rxApp.js]: fixed.

**Pull request merged**